yt_dlp/extractor/archiveorg.py

```diff
@@ -945,3 +947,240 @@ def _real_extract(self, url):
+            'creator': 'Dispatch',
+            'view_count': int,
+            'subtitles': 'mincount:6',
+            'uploader_id': 'V__FRA08071',
+            'uploader_url': 'http://vlive.tv',
+            'uploader': None,
+            'upload_date': '20181130',
+            'thumbnail': r're:^https?://.*\.(?:jpg|png)$',
+            'timestamp': 1543601327,
+            'like_count': int,
+            'channel': 'Dispatch',
+            'channel_id': 'C796F3',
+            'comment_count': int,
+            'release_timestamp': 1543601040,
+            'release_date': '20181130',
+            'duration': 279,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    # The wayback machine has special timestamp and "mode" values:
+    # timestamp:
+    #   1 = the first capture
+    #   2 = the last capture
+    # mode:
+    #   id_ = Identity - perform no alterations of the original resource, return it as it was archived.
+    _WAYBACK_BASE_URL = 'https://web.archive.org/web/2id_/'
+
+    def _download_wbm_page(self, url, video_id, timestamp='2', mode='id_', **kwargs):
+        for retry in self.RetryManager():
+            try:
+                return self._download_webpage(f'https://web.archive.org/web/{timestamp}{mode}/' + url, video_id, **kwargs)
+            except ExtractorError as e:
+                if isinstance(e.cause, urllib.error.HTTPError) and e.cause.code == 404:
+                    raise ExtractorError('Page was not archived', expected=True)
+                retry.error = e
+                continue
+
+    def _download_wbm_json(self, url, video_id, timestamp='2', mode='id_', **kwargs):
+        page = self._download_wbm_page(url, video_id, timestamp, mode, **kwargs)
+        if not page:
+            raise ExtractorError('Page was not archived', expected=True)
+        else:
+            return self._parse_json(page, video_id)
+
+    def _extract_formats_from_m3u8(self, m3u8_url, params, video_id):
+        m3u8_doc = self._download_wbm_page(m3u8_url, video_id, note='Downloading m3u8', query=params, fatal=False)
+        if m3u8_doc:
+            # M3U8 document is not valid, so it needs to be fixed
+            m3u8_doc_lines = m3u8_doc.splitlines()
+            modified_m3u8_doc_lines = []
+            url_base = m3u8_url.rsplit('/', 1)[0]
+            first_segment = None
+            for line in m3u8_doc_lines:
+                if line.startswith('#'):
+                    modified_m3u8_doc_lines.append(line)
+                else:
+                    modified_line = f'{self._WAYBACK_BASE_URL}{url_base}/{line}?{urllib.parse.urlencode(params)}'
+                    modified_m3u8_doc_lines.append(modified_line)
+                    if first_segment is None:
+                        first_segment = modified_line
+            modified_m3u8_doc = '\n'.join(modified_m3u8_doc_lines)
+
+            # Segments may not have been archied. See 101870
```

Comment:
What's `101870`?

Reply:
An affected video ID. Having the full URL in the comment should make it clearer.